src/wire/tcp.rs

```diff
@@ -10,7 +10,7 @@ use super::ip::checksum;
 ///
 /// A sequence number is a monotonically advancing integer modulo 2<sup>32</sup>.
 /// Sequence numbers do not have a discontiguity when compared pairwise across a signed overflow.
-#[derive(Debug, PartialEq, Eq, Clone, Copy, Default)]
+#[derive(Debug, PartialEq, Eq, Ord, Clone, Copy, Default)]
```

Comment:
No no no, this is very intentional! Sequence numbers are **not** totally ordered! They wrap!

Reply:
Ah, yes. Then by just looking at the two numbers there is no way to distinguish between the case of normal operation with local_seq_no just wrapped before the max statement and the retransmission case. There is also a wrapped case for retransmission where the minimum would have to be taken. So first a way to find out if the value of  remote_last_seq was coming from a retransmission is needed, then it could be updated directly and there is no min/max involved.